src/librustc_mir/build/matches/mod.rs

```diff
@@ -1652,11 +1652,6 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             // denotes *R.
             let ref_for_guard =
                 self.storage_live_binding(block, binding.var_id, binding.span, RefWithinGuard);
-            // Question: Why schedule drops if bindings are all
-            // shared-&'s?
-            // Answer: Because schedule_drop_for_binding also emits
-            // StorageDead's for those locals.
-            self.schedule_drop_for_binding(binding.var_id, binding.span, RefWithinGuard);
```

Comment:
What about this comment no longer applies?

Reply:
`storage_live_binding` schedules `StorageDead` and it has done for some time.

Reply:
(Since PR #52046, specifically, it seems.)